pkg/windows/windows.go

```diff
@@ -534,28 +534,11 @@ func (vm *windows) ConfigureAzureCloudNodeManager(nodeName string) error {
 	return nil
 }
 
-func (vm *windows) ConfigureKubeProxy(nodeName, hostSubnet string) error {
-	endpointIP, err := vm.Run(NetworkConfScriptPath, true)
-	if err != nil {
-		return errors.Wrap(err, "error creating HNS endpoint")
-	}
-
-	kubeProxyServiceArgs := "--windows-service --v=4 --proxy-mode=kernelspace --feature-gates=WinOverlay=true " +
-		"--hostname-override=" + nodeName + " --kubeconfig=c:\\k\\kubeconfig " +
-		"--cluster-cidr=" + hostSubnet + " --log-dir=" + kubeProxyLogDir + " --logtostderr=false " +
-		"--network-name=" + OVNKubeOverlayNetwork + " --source-vip=" + strings.TrimSpace(endpointIP) +
-		" --enable-dsr=false"
-
-	kubeProxyService, err := newService(KubeProxyPath, kubeProxyServiceName, kubeProxyServiceArgs,
-		[]string{HybridOverlayServiceName})
-	if err != nil {
-		return errors.Wrapf(err, "error creating %s service object", kubeProxyServiceName)
-	}
-
-	if err := vm.ensureServiceIsRunning(kubeProxyService); err != nil {
-		return errors.Wrapf(err, "error ensuring %s Windows service has started running", kubeProxyServiceName)
+func (vm *windows) ConfigureKubeProxy() error {
+	vm.log.V(1).Info("waiting for kube-proxy to start running")
+	if err := vm.waitForServiceToRun(KubeProxyServiceName); err != nil {
```

Comment:
Why is this wait needed? Were we doing this in the past?

Reply:
That's explained in the commit message

Reply:
I think you should call it out in a comment here too. Is there a better way to figure this out? Maybe look at the node object?

Reply:
Commented. I don't know of a better way. This will be removed ASAP, added a TODO with a ticket.